File: cadwiki-nuget/cadwiki.AutoCAD2021.Base.Utilities/Shared/ObjectIds.vb
Imports Autodesk.AutoCAD.ApplicationServices
Imports Autodesk.AutoCAD.DatabaseServices

Public Class ObjectIds
    Public Shared Function GetEntity(doc As Document, objectId As ObjectId) As Entity
        Dim deletedEntities As New List(Of Entity)
        Dim db As Database = doc.Database
        Using lock As DocumentLock = doc.LockDocument()
            Using t As Transaction = db.TransactionManager.StartTransaction()
                Dim entity As Entity = CType(t.GetObject(objectId, OpenMode.ForRead), Entity)
                Return entity
            End Using
        End Using
    End Function
End Class

File: cadwiki-nuget/cadwiki.AC/Utilities/ObjectIds.cs
using System.Collections.Generic;
using Autodesk.AutoCAD.ApplicationServices;
using Autodesk.AutoCAD.DatabaseServices;

namespace cadwiki.AC.Utilities
{

    public class ObjectIds
    {
        public static Entity GetEntity(Document doc, ObjectId objectId)
        {
            var deletedEntities = new List<Entity>();
            var db = doc.Database;
            using (var @lock = doc.LockDocument())
            {
                using (var t = db.TransactionManager.StartTransaction())
                {
                    Entity entity = (Entity)t.GetObject(objectId, global::Autodesk.AutoCAD.DatabaseServices.OpenMode.ForRead);
                    return entity;
                }
            }
        }
    }
}